tests/unittests/tests-asan_canary/tests-asan.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2024 Christian Amsüss
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
```

Comment:
```suggestion
 * @file
 * @brief     Provide a true-positive (failing) test case for ASAN
```

Reply:
This is only the "DO NOT MERGE" that will be removed once verified -- that commit will be removed again because AFAICT we don't have infrastructure for required-to-fail-to-build tests.

Reply:
Bold words for someone working that late on the weekend ;)